### ISSUE TITLE
Correct the mentioned pylint errors to get perfect 10. (broad-except, too-many-nested-blocks, inconsistent-return-statements,, useless-else-on-loop, too-many-nested-blocks, inconsistent-return-statements)

### ISSUE THREAD

**Issue by LordGameleo** [xbmc/addon-check#185]
We are currently using a customized checklist for pylint. We have disabled a lot of checks to make our build successfully. We want to counter these checks few at a time. This issue deals with one of them.

## How to produce:

After forking and cloning this repo, make a branch named develop-Pylint_Issue#185 using
``git branch develop-Pylint_Issue#185``

Checkout branch by
``git checkout develop-Pylint_Issue#185``

remove  ```broad-except, too-many-nested-blocks```, ```inconsistent-return-statements```, ```useless-else-on-loop```, ```too-many-nested-blocks``` and  ```inconsistent-return-statements``` from .pylintrc file 

Check pylint errors using 
``pylint *`` on terminal (in the repo)

Add doc strings to get a perfect 10 score on pylint.

**Comment by shrawnz** [comment]
Hi, I'd like to pick up this issue

**Comment by LordGameleo** [comment]
@shrawnz  Yeah feel free to work on this one. Remember to deal with one of these check at once. Separate different checks by using separate commits.
